lib/rgeo/impl_helper/basic_polygon_methods.rb

```diff
@@ -79,8 +79,45 @@ def coordinates
         ([@exterior_ring] + @interior_rings).map(&:coordinates)
       end
 
+      def contains?(rhs)
+        if Feature::Point === rhs
+          contains_point?(rhs)
+        else
+          raise(Error::UnsupportedOperation,
+            "Method Polygon#contains? is only defined for Point"
+          )
+        end
+      end
+
       private
 
+      def contains_point?(point)
+        ring_encloses_point?(@exterior_ring, point) &&
+          !@interior_rings.any? do |exclusion|
+            ring_encloses_point?(exclusion, point, on_border_return: true)
+          end
+      end
+
+      def ring_encloses_point?(ring, point, on_border_return: false)
+        # This is an implementation of the ray casting algorithm, greatly inspired
+        # by https://wrf.ecse.rpi.edu/Research/Short_Notes/pnpoly.html
+        # Since this algorithm does not handle point on edge, we check first if
+        # the ring is on the border.
+        # on_border_return is used for exclusion ring
+        return on_border_return if ring.contains?(point)
```

Comment:
I think that we should get rid of this optional hash parameter `on_border_return` and just make it false in the code. We can always change it if we determine that should be true later on or some optional parameter, but as it stands this isn't configurable by the user so I think it's cleaner just to make the statement:

```rb
return false if ring.contains?(point)
```

and get rid of the parameter.

Reply:
This parameter is used to change the border behavior for exclusion rings. I'm not sure how we could remove it.

Reply:
Apologies, you're correct. I somehow missed that it was used with both `true` and `false`. 

This looks good then.